iml-agent/src/action_plugins/kernel_module.rs

```diff
@@ -0,0 +1,9 @@
+use crate::agent_error::ImlAgentError;
+
+use kmod;
+
+pub async fn loaded(module: String) -> Result<bool, ImlAgentError> {
+    let ctx = kmod::Context::new()?;
```

Comment:
I haven't looked very deeply at what happens in kmod_sys for this, but if it does any sort of work it should be inside a `run` call so it's in a separate blocking thread.